src/catalog/src/memory/objects.rs

```diff
@@ -1609,9 +1615,11 @@ impl CatalogItem {
             CatalogItem::Log(_) => {}
             CatalogItem::Table(_) => {}
             CatalogItem::Type(_) => {}
-            CatalogItem::View(view) => uses.extend(view.raw_expr.depends_on()),
-            CatalogItem::MaterializedView(mview) => uses.extend(mview.raw_expr.depends_on()),
-            CatalogItem::ContinualTask(ct) => uses.extend(ct.raw_expr.depends_on()),
```

Comment:
Do we ever have to worry about the `raw_expr` changing and if so are we correctly updating `dependencies`?

Reply:
I don't think so, doing manual modifications to the `raw_expr` feels like code smell? The only place I found that we changed the `raw_expr` was in continual tasks [here](https://github.com/MaterializeInc/materialize/blob/main/src/adapter/src/continual_task.rs#L48-L56) which needs to do it because it's the first type (AFAIK) that is self-referential, we could probably refactor this away by introducing a "self" type in name resolution instead of [using `LocalId` like we do today](https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/names.rs#L1657-L1659)